tests/test_config.py

```diff
@@ -26,6 +28,18 @@ def test_load_ssl_config_verify_existing_file():
     assert context.check_hostname is True
 
 
+@pytest.mark.parametrize("config", ("SSL_CERT_FILE", "SSL_CERT_DIR"))
+def test_load_ssl_config_verify_env_file(cert_pem_file, config):
+    os.environ[config] = (
+        cert_pem_file if config.endswith("_FILE") else str(Path(cert_pem_file).parent)
+    )
+    ssl_config = httpx.SSLConfig(trust_env=True)
+    context = ssl_config.load_ssl_context()
+    assert context.verify_mode == ssl.VerifyMode.CERT_REQUIRED
+    assert context.check_hostname is True
+    assert ssl_config.verify == os.environ[config]
```

Comment:
Verify that certs are loaded via `assert len(context.get_ca_certs) > 0`

Reply:
`get_ca_certs` returns empty if use `context.load_verify_locations(capath=str(ca_bundle_path))`. 
To get a result, we should create an `SSLSocket` and call [getpeercert](https://docs.python.org/3/library/ssl.html#ssl.SSLSocket.getpeercert) like below

```
if config == "SSL_CERT_DIR":
        HOST = "example.org"
        PORT = 443
        conn: ssl.SSLSocket = ssl_config.ssl_context.wrap_socket(
            socket.socket(socket.AF_INET, socket.SOCK_STREAM), server_hostname=HOST
        )
        conn.connect((HOST, PORT))
        conn.getpeercert()
```

Reply:
Maybe we can use https_server.url.host and https_server.url.port to accomplish that. Then our certificate will verify as well :)

Reply:
I couldn't verify with `cert_pem_file` for `https_server.url`. Please help :)

Reply:
@cansarigol Is there anything we can do to help unblock this?

Reply:
Would be great. Thanks.  I'm stuck about create a sslsocket because of ssl validation error. I don't know if it is related trustme scope

Reply:
@cansarigol Can you share a traceback so we know what part is failing exactly? :) Thanks

Reply:
I'm very sorry that I didn't do anything but i will asap

Reply:
This is blocked by #354, just gave it a spin myself. I think we should maybe merge this now?

Reply:
#354 is closed now, I tried changing the test to look like this right now:

```python
    os.environ[config] = (
        ca_cert_pem_file
        if config.endswith("_FILE")
        else str(Path(ca_cert_pem_file).parent)
    )
    ssl_config = httpx.SSLConfig(trust_env=True)
    context = ssl_config.load_ssl_context()
    assert context.verify_mode == ssl.VerifyMode.CERT_REQUIRED
    assert context.check_hostname is True
    assert ssl_config.verify == os.environ[config]

    host = https_server.url.host
    port = https_server.url.port
    conn = socket.create_connection((host, port))
    context.wrap_socket(conn, server_hostname=host)
    assert len(context.get_ca_certs()) == 1
```

But I'm still failing on the `SSL_CERT_DIR` test case and I don't know why, pretty annoying!

Reply:
I added a test to be able to compare with below traceback.

```
tests/test_config.py ...FSSL error in data received
protocol: <asyncio.sslproto.SSLProtocol object at 0x103c044a8>
transport: <_SelectorSocketTransport closing fd=19 read=idle write=<idle, bufsize=0>>
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/sslproto.py", line 526, in data_received
    ssldata, appdata = self._sslpipe.feed_ssldata(data)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/sslproto.py", line 189, in feed_ssldata
    self._sslobj.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 763, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1045)
```

Reply:
Spoke with @florimondmanca and I think we're just going to skip the `SSL_CERT_DIR` case here and if we run into issues with it we'll fix them then. Until then this PR is good to go. Thank you so much @cansarigol for following this one all the way through. :)